site/content/3.12/release-notes/version-3.12/incompatible-changes-in-3-12.md

```diff
@@ -136,6 +136,23 @@ changed in the documents of SmartGraph vertex collections. This is now strictly
 See [API Changes in ArangoDB 3.12](api-changes-in-3-12.md#validation-of-smartgraphattribute-in-smartgraphs)
 for details and instructions on how to repair affected attributes.
 
+## Prevent dropping a collection from a graph
+
+Dropping collections that are part of an existing graph is now strictly enforced
+so that the graph definition remains intact even if you attempt to drop a
+collection.
+
+While previously it was allowed to drop collections that were part of an existing
+graph, trying to do so results in the following error:
+`ERROR_GRAPH_MUST_NOT_DROP_COLLECTION` with the `1942` error code.
+
+This may require changes in the client application code that drops individual
+collections from graphs for clean-up purposes. The recommended way to drop
+these collections is to drop the graph instead. Dropping the graph can optionally
+delete all associated collections of the graph as well. Even if this is not used,
+collections that are not part of a graph definition anymore can later be dropped
+regularly.
```

Comment:
Should we describe the other workflow?
- You can delete an entire relation from the graph definition (although the web interface doesn't let you do that with the first relation)
- You can unset the edge collection of a relation (even though the resulting definition seems invalid but is apparently possible)
- You can remove vertex collections from a relation, which moves them to orphan collections. Orphan collections can be removed from the graph definition, and then you can drop these collections

Reply:
I suggest we recommend
- remove a vertex collection from the edge definitions
- remove the orphan collection from the graph using the graph API, which has a `dropCollection` option (as opposed to dropping the collection separately with the document API)